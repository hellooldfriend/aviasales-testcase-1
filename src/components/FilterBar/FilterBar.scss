.filter_bar {
    display: inline-block;
    width: 100%;
    margin-bottom: 20px;
    white-space: nowrap;

    button {
        text-transform: uppercase;
        font-size: 12px;
        line-height: 20px;
        letter-spacing: 0.55px;
        border-radius: 0;
        padding: 15px 25px;

        &:first-child {
            border-radius: 6px 0 0 6px;
        }

        &:last-child {
            border-radius: 0 6px 6px 0;
        }
    }
}