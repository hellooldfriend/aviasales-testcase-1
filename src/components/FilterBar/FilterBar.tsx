import React, { useState } from 'react'
import './FilterBar.scss'

import Button from '../Button'


const OPTIONS = [
    {
        code: 'cheap',
        title: 'Самый дешёвый',
    },
    {
        code: 'fast',
        title: 'Самый быстрый',
    },
    {
        code: 'optimum',
        title: 'Оптимальный',
    },

]

const FilterBar = () => {
    const [currentOption, setCurrentOption] = useState<null | string>(null)


    return (
        <div className="filter_bar">
            {OPTIONS.map(option => {
                const isActive = currentOption ? currentOption === option.code : false
                return (
                    <Button
                        key={option.code}
                        active={isActive}
                        value={option.title}
                        onClick={() => setCurrentOption(option.code)}
                    />
                )
            })}
        </div>
    )
}


export default FilterBar