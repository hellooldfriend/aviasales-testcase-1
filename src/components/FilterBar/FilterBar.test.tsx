import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

import FilterBar from './FilterBar'


afterEach(() => {
    cleanup()
})

const props = {
    mode: null,
    onChange: () => {},
}


describe('FilterBar', () => {
    test('Should render FilterBar', () => {
        render(<FilterBar {...props} />)
        const element = screen.getByTestId('filter_bar')
        expect(element).toBeInTheDocument()
    })

    test('Matches snapshot', () => {
        const element = renderer.create(<FilterBar {...props} mode={'cheap'} />).toJSON()
        expect(element).toMatchSnapshot()
    })
})
