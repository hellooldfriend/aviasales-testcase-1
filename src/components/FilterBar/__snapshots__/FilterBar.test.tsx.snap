// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar Matches snapshot 1`] = `
<div
  className="filter_bar"
  data-testid="filter_bar"
>
  <button
    className="button _active"
    data-testid="button"
    onClick={[Function]}
  >
    Самый дешёвый
  </button>
  <button
    className="button"
    data-testid="button"
    onClick={[Function]}
  >
    Самый быстрый
  </button>
  <button
    className="button"
    data-testid="button"
    onClick={[Function]}
  >
    Оптимальный
  </button>
</div>
`;
